src/macros.jl

```diff
@@ -465,45 +462,98 @@ macro constraint(args...)
     end)
 end
 
+# This function needs to be implemented by all `AbstractModel`s
+constrainttype(m::Model) = ConstraintRef{typeof(m)}
 
 """
-    @SDconstraint(m, x)
-
-Adds a semidefinite constraint to the `Model m`. The expression `x` must be a square, two-dimensional array.
+    @constraint(m::Model, expr)
+
+Add a constraint described by the expression `expr`.
+
+    @constraint(m::Model, ref[i=..., j=..., ...], expr)
+
+Add a group of constraints described by the expression `expr` parametrized by
+`i`, `j`, ...
+
+The expression `expr` can either be
+
+* of the form `func in set` constraining the function `func` to belong to the
+  set `set`, e.g. `@constraint(m, [1, x-1, y-2] in MOI.SecondOrderCone)`
```

Comment:
Currently this needs to be `MOI.SecondOrderCone(3)`, right?

Reply:
Ups, yes indeed !